systemverilog-plugin/UhdmAst.cc

```diff
@@ -3408,120 +3477,171 @@ void UhdmAst::process_operation(const UHDM::BaseClass *object)
 
 void UhdmAst::process_stream_op()
 {
-    // Create a for loop that does what a streaming operator would do
-    auto block_node = find_ancestor({AST::AST_BLOCK, AST::AST_ALWAYS, AST::AST_INITIAL});
-    auto process_node = find_ancestor({AST::AST_ALWAYS, AST::AST_INITIAL});
-    auto module_node = find_ancestor({AST::AST_MODULE, AST::AST_FUNCTION, AST::AST_PACKAGE});
-    log_assert(module_node);
-    if (!process_node) {
-        if (module_node->type != AST::AST_FUNCTION) {
-            // Create a @* always block
-            process_node = make_ast_node(AST::AST_ALWAYS);
-            module_node->children.push_back(process_node);
-            block_node = make_ast_node(AST::AST_BLOCK);
-            process_node->children.push_back(block_node);
-        } else {
-            // Create only block
-            block_node = make_ast_node(AST::AST_BLOCK);
-            module_node->children.push_back(block_node);
-        }
+    // Closest ancestor where new statements can be inserted.
+    AST::AstNode *stmt_list_node = find_ancestor({
+      AST::AST_MODULE,
+      AST::AST_PACKAGE,
+      AST::AST_BLOCK,
+      AST::AST_INITIAL,
+      AST::AST_ALWAYS,
+      AST::AST_FUNCTION,
+    });
+    log_assert(stmt_list_node != nullptr);
```

Comment:
Please change this to ``log_file_error``, that logs also filename and line number as we might sometimes hit this assert.

Reply:
I see the point, but to not mix assertions with errors I've added yet another thing to this PR: `uhdmast_assert` (see commits).
It prints errors like:
```
/yosys-systemverilog/yosys-f4pga-plugins/systemverilog-plugin/UhdmAst.cc:3536: error: Assertion failed: stmt_list_node != nullptr
/yosys-systemverilog/yosys-f4pga-plugins/systemverilog-plugin/UhdmAst.cc:3536: note: In function: void systemverilog_plugin::UhdmAst::process_stream_op()
/yosys-systemverilog/UHDM-integration-tests/tests/StreamOpInitialAlways/dut.sv:2018: note: When processing object of type 'operation'.
```

Ofc I've used it here and in the rest of the method.